crates/sui-types/Cargo.toml

```diff
@@ -42,9 +42,9 @@ move-command-line-common.workspace = true
 move-core-types.workspace = true
 move-disassembler.workspace = true
 move-ir-types.workspace = true
-move-vm-test-utils.workspace = true
-move-vm-types.workspace = true
-move-vm-profiler.workspace = true
+move-vm-test-utils = { workspace = true, features = ["gas-profiler"] }
+move-vm-types = { workspace = true, features = ["gas-profiler"]}
+move-vm-profiler = { workspace = true, features = ["gas-profiler"] }
```

Comment:
Do these need to be set here?  IIRC Cargo feature rules this should be set by the `gas-profiler` feature below.

Reply:
This imports the feature declared by this dependency crate. I think the question you meant to ask was, does the workspace-hack dependency section declaration of the features prevent us from needing to add these here? The answer is, oddly no. If we were to remove these and then run `cargo hakari explain [dep name]` we would see that it automatically builds additional versions of the project with all possible features enabled, which is not what we want. I can't really explain why, because that also isn't documented anywhere.

Reply:
I have the same question as @tzakian (as originally posed):

The `gas-profiler` feature for this crate, defined in the `[features]` section below implies the `gas-profiler` features for these crates here, and I would have expected that for that to do anything, you would need to not include the `gas-profiler` feature explicitly here (as suggested by the Cargo docs on [Dependency Features](https://doc.rust-lang.org/cargo/reference/features.html#dependency-features))

My expectation of how this configuration pans out is that it results in `gas-profiler` always being enabled in these crates, and because `sui-types` is such a foundational crate, that makes me think that it's going to end up enabled in `sui-node`.

I'm not a `hakari` expert though, so I can't speak to that bit, but I'll do some digging after this review to convince myself that this has the desired effect (namely that when we build `sui-node` alone, we don't include gas-profiler things).

Reply:
Good news, I tried one more thing and was able to remove all these `featues = [ "gas-profiler"]`  from the dependencies. They would have been turning on subsets of the several feature flags, which as far as I understand, we are still doing because of the contents of workspace-hack/Cargo.toml . It was ok either way, because having some of those features turned on would just expose functions that were unused, since the source feature that acts as the root which enables all is in the sui crate, and so we had like small sub-trees of features being enabled, which did nothing on their own, since nothing was calling them.